packages/discord.js/typings/index.d.ts

```diff
@@ -5006,20 +5010,20 @@ export interface Caches {
   // TODO: GuildManager: [manager: typeof GuildManager, holds: typeof Guild];
   GuildMemberManager: [manager: typeof GuildMemberManager, holds: typeof GuildMember];
   GuildBanManager: [manager: typeof GuildBanManager, holds: typeof GuildBan];
-  GuildForumThreadManager: [manager: typeof GuildForumThreadManager, holds: typeof ThreadChannel<true>];
+  GuildForumThreadManager: [manager: typeof GuildForumThreadManager, holds: ForumThreadChannel];
   GuildInviteManager: [manager: typeof GuildInviteManager, holds: typeof Invite];
   GuildMessageManager: [manager: typeof GuildMessageManager, holds: typeof Message<true>];
   GuildScheduledEventManager: [manager: typeof GuildScheduledEventManager, holds: typeof GuildScheduledEvent];
   GuildStickerManager: [manager: typeof GuildStickerManager, holds: typeof Sticker];
-  GuildTextThreadManager: [manager: typeof GuildTextThreadManager, holds: typeof ThreadChannel<false>];
+  GuildTextThreadManager: [manager: typeof GuildTextThreadManager, holds: TextThreadChannel];
   MessageManager: [manager: typeof MessageManager, holds: typeof Message];
   // TODO: PermissionOverwriteManager: [manager: typeof PermissionOverwriteManager, holds: typeof PermissionOverwrites];
   PresenceManager: [manager: typeof PresenceManager, holds: typeof Presence];
   ReactionManager: [manager: typeof ReactionManager, holds: typeof MessageReaction];
   ReactionUserManager: [manager: typeof ReactionUserManager, holds: typeof User];
   // TODO: RoleManager: [manager: typeof RoleManager, holds: typeof Role];
   StageInstanceManager: [manager: typeof StageInstanceManager, holds: typeof StageInstance];
-  ThreadManager: [manager: typeof ThreadManager, holds: typeof ThreadChannel];
+  ThreadManager: [manager: typeof ThreadManager, holds: AnyThreadChannel];
```

Comment:
My hunch was correct: this creates a discrepancy between what the makeCache method gets (manager, class, manager) and what the types imply when a threadmanager is brought into the mix (which implies the class is an instance). This still needs a typeof in some form

Reply:
Just reverted the changes here since there's only ever the one constructor at runtime.